packages/requisitions/src/SupplierRequisition/DetailView/DetailView.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { FC } from 'react';
+import { useParams } from 'react-router';
+import {
+  TableProvider,
+  createTableStore,
+  useOmSupplyApi,
+  useDocument,
+  useColumns,
+  GenericColumnKey,
+  DataTable,
+  usePagination,
+  useTranslation,
+} from '@openmsupply-client/common';
+
+import { reducer } from './reducer';
+import { getSupplierRequisitionDetailViewApi } from './api';
+import { Toolbar } from './Toolbar';
+import { Footer } from './Footer';
+import { AppBarButtons } from './AppBarButtons';
+import { SidePanel } from './SidePanel';
+
+import { isRequisitionEditable } from '../../utils';
+import { RequisitionLine } from '../../../../mock-server/src/data';
+
+const useDraftSupplierRequisition = () => {
+  const { id } = useParams();
+  const { api } = useOmSupplyApi();
+
+  const { draft, save, dispatch, state } = useDocument(
+    ['requisition', id],
+    reducer,
+    getSupplierRequisitionDetailViewApi(api)
+  );
+
+  const onChangeSortBy = () => {};
+
+  return { draft, save, dispatch, onChangeSortBy, sortBy: state.sortBy };
+};
+
+export const DetailView: FC = () => {
+  const { draft, save, onChangeSortBy, sortBy } = useDraftSupplierRequisition();
+
+  const columns = useColumns<RequisitionLine>(
+    [
+      ['itemCode', { width: 50 }],
+      ['itemName', { width: 150 }],
+
+      'monthlyConsumption',
+
+      [
+        'previousStockOnHand',
+        {
+          accessor: line =>
+            `${line.previousStockOnHand} (${Math.floor(
+              line.previousStockOnHand / line.monthlyConsumption
+            )} months)`,
+        },
+      ],
+      [
+        'calculatedQuantity',
+        {
+          accessor: line => {
+            const threeMonthsStock = line.monthlyConsumption * 3;
+            const diff = threeMonthsStock - line.previousStockOnHand;
+            if (diff > 0) {
+              return `${diff.toFixed(2)} (${Math.floor(
+                diff / line.monthlyConsumption
+              )} months)`;
+            } else {
+              return 0;
+            }
```

Comment:
made an assumption/placeholder that the 'target' is 3 months worth of stock

Reply:
it's the `max MOS` field (now MOS) which isn't shown yet. defaults to `3` - but is a property of the requisition

Reply:
Of course, that makes sense!